### PR TITLE
Finished SendImageInteractor and SendImagePresenter.

### PR DESCRIPTION
**Pull request by teamagam3**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/103" target="_blank">ReviewNinja

File: data/src/main/java/com/teamagam/gimelgimel/data/images/ImagesDataRepository.java [hunk 1]
```diff
@@ -29,11 +34,14 @@ public ImagesDataRepository(GGMessagingAPI api, MessageDataMapper mapper) {
...
-    public void uploadImage(MessageImage message, String fileName, byte[] imageBytes) {
+    public Observable<MessageImage> uploadImage(MessageImage message, String filePath) {
         MessageData messageData = mMessageMapper.transformToData(message);
-        MultipartBody.Part body = createMultipartBody(fileName, imageBytes);
+        File imageFile = new File(filePath);
+        byte[] compressedImageBytes = compressImage(imageFile);
+        MultipartBody.Part body = createMultipartBody(imageFile.getName(), compressedImageBytes);
 
-        mApi.sendImage(messageData, body);
+        return mApi.sendImage(messageData, body).map(returnedMessage ->
```
Comment (teamagam1) [inline data/src/main/java/com/teamagam/gimelgimel/data/images/ImagesDataRepository.java]: .map(... in new linew

File: data/src/main/java/com/teamagam/gimelgimel/data/images/ImagesDataRepository.java [hunk 2]
```diff
@@ -60,4 +68,37 @@ public void uploadImage(MessageImage message, String fileName, byte[] imageBytes
 
         return MultipartBody.Part.createFormData(Constants.IMAGE_KEY, fileName, requestFile);
     }
+
+    private byte[] compressImage(File image) {
```
Comment (teamagam1) [inline data/src/main/java/com/teamagam/gimelgimel/data/images/ImagesDataRepository.java]: consider extracting method(s) to ImageUtils

File: domain/build.gradle [hunk 3]
```diff
@@ -14,4 +14,5 @@ dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     compile 'io.reactivex:rxjava:1.1.8'
     compile 'com.google.code.gson:gson:2.6.2'
+    compile 'javax.inject:javax.inject:1'
```
Comment (teamagam1) [inline domain/build.gradle]: where do you use inject in the domain?

File: domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    public SendImageMessageInteractor(ThreadExecutor threadExecutor, PostExecutionThread postExecutionThread,
+                                      ImagesRepository repository) {
+        super(threadExecutor, postExecutionThread);
+
+        mImagesRepository = repository;
+    }
+
+    public void sendImageMessage(Subscriber subscriber, MessageImage messageImage, String imagePath) {
+        if (isValid(subscriber, messageImage, imagePath)) {
```
Comment (teamagam1) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java:29]: why do you need this check?

File: domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            mImagePath = imagePath;
+
+            execute(subscriber);
+        } else {
+            throw new IllegalArgumentException("None of the arguments can be null");
+        }
+    }
+
+    private boolean isValid(Subscriber subscriber, MessageImage messageImage, String imagePath) {
+        return !(subscriber == null || messageImage == null || imagePath.isEmpty());
```
Comment (teamagam1) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java]: imagePath == null?

File: domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Override
+    protected Observable buildUseCaseObservable() {
+        return mImagesRepository.uploadImage(mMessageImage, mImagePath);
```
Comment (teamagam1) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java]: we need also to put the message in the message repository

File: presentation/src/main/java/com/teamagam/gimelgimel/presentation/presenters/SendImageMessagePresenter.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    public void destroy() {
+        mView.hideProgress();
+    }
+
+    public void sendImageMessage(String senderId, String imagePath, PointGeometry geometry) {
+        mView.showProgress();
+
+        Subscriber subscriber = new SendImageMessageSubscriber();
+        ImageMetadata metadata =
+                new ImageMetadata(System.currentTimeMillis(), imagePath, geometry, "User");
```
Comment (teamagam1) [inline presentation/src/main/java/com/teamagam/gimelgimel/presentation/presenters/SendImageMessagePresenter.java]: use the field above

File: presentation/src/main/java/com/teamagam/gimelgimel/presentation/presenters/SendImageMessagePresenter.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        MessageImage message = new MessageImage(senderId, metadata);
```
Comment (teamagam1) [inline presentation/src/main/java/com/teamagam/gimelgimel/presentation/presenters/SendImageMessagePresenter.java:55]: we need to talk about it. I've created prerences.getUserName interface for the presenter

File: domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java (hunk 4 above)
Comment (teamagam3) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java:29]: It's just like what you did in SendMessagePresenter, you checked if the message was null before creating the observable.

File: domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java (hunk 5 above)
Comment (teamagam3) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java]: If the imagePath == null it's NOT valid.
The method returns true if the input valid.

**Pull request merged**